test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,564 @@
+import java.lang.management.ManagementFactory;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.management.JMX;
+import javax.management.MBeanServer;
+import javax.management.MBeanServerConnection;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import com.ibm.lang.management.*;
+import openj9.lang.management.*;
+import com.ibm.jvm.InvalidDumpOptionException;
+import com.ibm.jvm.DumpConfigurationUnavailableException;
+
+/**
+ * Class for testing the APIs that are provided as part of OpenJ9DiagnosticsMXBean
+ * Interface
+ *
+ */
+
+@Test(groups = { "level.extended" })
+public class TestOpenJ9DiagnosticsMXBean {
+	private static Logger logger = Logger.getLogger(TestOpenJ9DiagnosticsMXBean.class);
+	private static Process remoteServer;
+	private static Watchdog watchdog;
+	private ObjectName mxbeanName = null;
+	private OpenJ9DiagnosticsMXBean diagBean = null;
+	List<String> initialDumpOptions = new ArrayList<String>();
+
+	@BeforeClass
+	public void setUp() throws Exception {
+		/* We need the object name in any case - remote as well as local. */
+		try {
+			mxbeanName = new ObjectName("openj9.lang.management:type=OpenJ9Diagnostics");
+		} catch (MalformedObjectNameException e) {
+			e.printStackTrace();
+			Assert.fail("MalformedObjectNameException!");
+		}
+
+		initialDumpOptions = getDumpOptions();
+		
+	}
+
+	/**
+	 * Function to obtain the dump options 
+	 *
+	 */
+	private List<String> getDumpOptions() {
+		List<String> list = new ArrayList<String>();
+		try {
+			ProcessBuilder builder = new ProcessBuilder("java", "-Xdump:what", "-version");
+			Process process = builder.start();
+			builder.redirectErrorStream(true);
+
+			BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+			String line;
+			while(null != (line = reader.readLine())) {
+				logger.info(line);
+				list.add(line);
+			}
+		} catch (Exception e){
+			Assert.fail("Unexpected Exception occurred");
+		}
+		return list;
+	}
+
+	@Test
+	public void testLocal() {
+		MBeanServer mbeanServer = null;
+
+		try {
+			mbeanServer = ManagementFactory.getPlatformMBeanServer();
+			if (true != mbeanServer.isRegistered(mxbeanName)) {
+				Assert.fail("OpenJ9DiagnosticsMXBean is not registered. Cannot Proceed.");
+			}
+
+			diagBean = JMX.newMXBeanProxy(mbeanServer, mxbeanName, OpenJ9DiagnosticsMXBean.class);
+		} catch (Exception e) {
+			Assert.fail("Exception occurred in setting up local environment.");
+		} 
+	
+		/* Test OpenJ9DiagnosticsMXBean APIs */	
+		test_triggerDump(diagBean);
+
+		test_triggerDumpToFile(diagBean, "local");
+
+		test_triggerClassicHeapDump(diagBean);
+
+		test_resetDumpOptions(diagBean);
+
+		test_setDumpOptions(diagBean, "local");
```

Comment:
These would be better captured as 5 separate tests unless there is a reason not to do so.

Reply:
These 5 methods are reused in both testLocal() and testRemote() hence haven't split them as separate tests.  If it needs to be split that way then I can add two files - TestOpenJ9DiagnosticsMXBeanLocal.java and TestOpenJ9DiagnosticsMXBeanRemote.java and make these as 5 tests.

Reply:
I meant those should be tests that can be run independently. So instead of two methods with the `@Test` annotation, there would be ten (five local and five remote). That way we can be more confident that the tests don't interfere with each other.